### DISCUSSION TITLE
Question regarding the generation of mutated peptide sequence

### DISCUSSION THREAD

**Discussion by ec-ho-ra-mos** [vdemichev/DiaNN#412] [Q&A]
Hello Professor Vadim! 

I have two questions regarding this statement from the DIA-NN paper: "_For each target precursor in the spectral library, a decoy precursor is generated, if not provided in the library. By default, this is done by replacing the fragment ion m/z values of the target precursor assuming the amino acids adjacent to the peptide termini were mutated (GAVLIFMPWSCTYHKRQEND to LLLVVLLLLTSSSSLLNDQE mutation pattern is used)_".

1. How exactly does the GAVLIFMPWSCTYHKRQEND to LLLVVLLLLTSSSSLLNDQE mutation pattern work? Can I ask for a detailed explanation/guide on how this mutation pattern works? For example, I have 3 peptides "PEPTIDEEE", "SCTYNQQQ", and "AAAAAAAEEEEEELLLL", how will these peptides be mutated according to this pattern?
2. By this statement, specifically the part about the fragment ion m/z, does this means that the fragment ion m/z from the mutated peptides are used instead of the supposed correct fragment ions for the target, is my understanding correct?

My rationale for this question is I want to generate decoys using the same pattern as the one from DIA-NN.

A similar discussion regarding this question was raised in [December 2021](https://github.com/vdemichev/DiaNN/discussions/255). But I don't think OP's question, nor mine, was answered sufficiently. 

Thank you very much for your attention,
Echo

**Comment by vdemichev** [comment]
Hi Echo,

1. Each amino acid adjacent to the termini is replaced by the respective replacement aa (e.g. G replaced with L and L replaced with V) using the pattern.
2. Yes.

Best,
Vadim

**Reply by ec-ho-ra-mos** [reply]
Hello Professor Vadim,

Thank you very much for your reply!
Echo
